### DISCUSSION TITLE
Distance assigment trigger point

### DISCUSSION THREAD

**Discussion by MrChillings** [AntiMicroX/antimicrox#897] [Q&A] [answered]
Hey all,

i have a Xbox Series X controller and i am trying to assign a set change based on the left trigger axis (distance).

But its not behaving the way i'd like to. 

lets say we want to trigger the set change on 20% distance. now what i expect is that the set change occurs when it reaches 20% but that is not the case. 

It actually trigger when i release the trigger and passes the 20% value (lets say you got to 25% and go back to 0%, the moment it registers 20% it triggers the set change)

Am i doing something wrong, is there another option to this?

**Comment by pktiuk** [comment]
So it triggers the change, when 20% is reached (19->20->21->...),    
but not when it is skipped (17->19->21->...)?

**Reply by MrChillings** [reply]
no when reaching distance 19->20->21->... nothing happens.
But when we decrease back  21 -> 20 it triggers the change.

it looks like it has to surpass the given % and then when you drop back to the given percentage it triggers the set change.

it does function normally with letters.

**Reply by MrChillings** [reply]
~i figured it out.~ ### **EDIT i did not. solution is not working**

So in the advanced assignment I tried to set it up as : [Distance ...%] -> [Set number]. [ looks more logical to me] 

~But for some reason distance set change requires the set number first and then the Distance ... %. see picture:~
![afbeelding](https://github.com/AntiMicroX/antimicrox/assets/154590517/18994790-ac6c-4008-9ffa-3ff3624a843c)
**This is not working, the second you release it no matter what distance triggers the first assigned set.**

it gets even more odd when you put it in a logical sequence like set 1-> distance % -> set 3. Then the result would be set 3

Other question:
Is it somehow possible to put multiple distances in one assignment? 
My idea is to enter (from set 1) set 2 when distance 20% has reached so the trigger is already over the 20% threshold then either you can release it and  trigger set 1 again after passing 20% or press further to  90% to trigger set 3.

example : (you entered  at 20% now holding 50%)  < 20% = triggerpoint | or | > 90% = triggerpoint

**Reply by pktiuk** [reply]
So I think this is a bug in the app itself.

**Reply by MrChillings** [reply, marked as answer]
I sort of found a solution to my problem 👍 :

oke so distance + change set is not working "correctly". You would think first setting a [distance] followed by [change set] would trigger at the set change based on the reach distance. 

**Current way how AntiMicroX handles this**
[Distance 50%]  -> [set change] (nothing happens)
but it does trigger when we surpass the distance and decline back to the given distance: 
50% (nothing) -> 51% (passed threshold) -> 50% (decline back to threshold) = Set change

**My "workaround"**
So I figured out if I put the set change in between 2 Distance values it does behave the way i want to. see image below.
![afbeelding](https://github.com/AntiMicroX/antimicrox/assets/154590517/48e13238-e6e1-4aa4-b3a2-eb4d5bd01db0)

so now when reach 10% = set change. ( i think the program sort of handles it like this: the first 10% is reached = reserve set change -> new if action block (the 10% again) = release reserved set change. ). as long as the second distance block is equal or lower than the first distance block it triggers the set change.

its also possible to que up multiple set changes for example: 
![afbeelding](https://github.com/AntiMicroX/antimicrox/assets/154590517/789fe118-e567-42f6-9bf0-36e64e2c7152)


in my case i either want to jump to set 2 or 3. now because you can flick the trigger it sort of skips the 10% registration. if you slowly presses the trigger it registers the 10%


**odd founding**
while testing around i found it odd that you can only add so many distance blocks untill the sum total of distance blocks is 100%. 
For example if your first distance block is [90%] your second cant be 50% ( the sum would be 140%). in this case the max distance for the second distance block would be [10%]. or 10 times 1% etc.

Or perhaps i am already using this function in a way it was not designed for, idk.
